packages/auth-passport/authService.js

```diff
@@ -26,53 +28,42 @@ PassportAuthService.prototype.init = function() {
 }
 
 /**
+ * TODO: update this description
  * Sends a request to the profile endpoint to retrieve the user's profile data.
  * @returns Returns the profile data retrieved from the server.
  */
 PassportAuthService.prototype.getProfile = function() {
-  var req = {
-    method: 'GET',
-    url: cloudUrl + CONSTANTS.PROFILE_URL
-  };
   var self = this;
-  return this.http(req, { withCredentials: true }).then(function(res) {
-    userProfile = res.data;
-    localStorage.setItem(USER_CACHE_KEY, JSON.stringify(userProfile));
-    return userProfile;
-  }).catch(function(err) {
-    if (err.status === 401) {
-      this.window.location = cloudUrl + CONSTANTS.LOGIN_URL;
-      return;
-    }
-    if (err.status === 403) {
-      self.dialog.show(self.dialog.alert({
-        title: 'Forbidden',
-        textContent: 'You are not authorized to access this resource.',
-        ok: 'OK'
-      }));
-      return;
-    }
-    var cachedUser;
-    try {
-      cachedUser = JSON.parse(localStorage.getItem(USER_CACHE_KEY));
-    } catch (err) {
+  if (self.isMobile) {
```

Comment:
If we can avoid `isMobile` check later that will be awesome. We can load different versions of authService to mobile and portal. Auth service is an class so we can just override one method and mount (typical object extend). isMobile is typical javascript/structural way of handling this.